File: app/src/main/java/android/montreal/devfest/stuff/presentation/presenters/MainPresenterModule.java
package android.montreal.devfest.stuff.presentation.presenters;

import android.montreal.devfest.stuff.presentation.navigation.Navigator;
import android.montreal.devfest.stuff.usecases.ManageRandomStuffUseCase;

import dagger.Module;
import dagger.Provides;

@Module
public abstract class MainPresenterModule {
    @Provides
    static MainPresenter presenter(
            MainPresenter.MainView view,
            ManageRandomStuffUseCase randomStuffUseCase,
            Navigator.ViewMore viewMore
    ) {
        return new MainPresenterImpl(view, randomStuffUseCase, viewMore);
    }
}

File: stuff_presentation/src/main/java/android/montreal/devfest/stuff/presentation/presenters/MainPresenterModule.kt
package android.montreal.devfest.stuff.presentation.presenters

import android.montreal.devfest.stuff.presentation.navigation.Navigator
import android.montreal.devfest.stuff.usecases.ManageRandomStuffUseCase

import dagger.Module
import dagger.Provides

@Module
abstract class MainPresenterModule {
    @Module
    companion object {
        @Provides
        @JvmStatic
        internal fun presenter(
                view: MainPresenter.MainView,
                randomStuffUseCase: ManageRandomStuffUseCase
        ): MainPresenter {
            return MainPresenterImpl(view, randomStuffUseCase)
        }
    }
}
